### PR TITLE
Feature: Single & Multiple sitemaps

### LINKED ISSUE #6: Option for seperated sitemaps?
Do I miss something or right now module allow me to make only one sitemap.xml and I can't create separated sitemaps for example for users, pages, products etc.?

<div align="right"><sub><em>This feature request is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt-community/sitemap-module/issues/c5">#c5</a>)</em></sub></div>

### LINKED ISSUE #19: Multiple sitemaps possible?
Hello. Is it possible to create separate multiple sitemaps?
I want separate sitemap for Google News.

<div align="right"><sub><em>This feature request is available on <a href="https://nuxtjs.cmty.io">Nuxt.js</a> community (<a href="https://nuxtjs.cmty.io/nuxt-community/sitemap-module/issues/c18">#c18</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by WilliamDASILVA**
Able to create a single sitemap like before by assignign a single object as the module option or pass an array of objects, each object being a single sitemap.

Switched from jest to ava for unit tests.

Fixes #6 and fixes #19

### CODE REVIEW

**Comment by WilliamDASILVA** [comment]
@NicoPennec Saw you added more tests in master, should I remove the migration to Ava?

**Comment by NicoPennec** [comment]
Thank you for this pull request. It will be an useful feature :)

But there are some points of divergence.
I wish to keep Jest instead of AVA, to follow the Nuxt ecosystem test suit. **(1)**
Besides, the minimum version of engine from package.json must be compliant with the current Nuxt.js requirement, so >=8.0.0 **(2)**
Then, Babel is not usefull with node 8.0.0, because it supports natively ES2016, so I try to keep a pure vanilla JS code base.

I'm working on a refactoring based on the new hooks from Nuxt >=1.x, that will be release very soon. It's the reason of my commit with new tests.

It’s planned to support multiple sitemaps. I should pick up your method for this update. Nice job


**(1)**
module template
https://github.com/nuxt-community/module-template
jest
https://github.com/nuxt/nuxt.js/blob/dev/package.json#L152
https://stackshare.io/stackups/jest-vs-ava (popularity)

**(2)**
node engine
https://github.com/nuxt/nuxt.js/blob/dev/package.json#L67

**Comment by Livog** [comment]
Would be awesome to see this feature as if you are doing a big project like myself. I can't have sitemaps that are bigger than 50MB or 50000 <loc> tags.

**Comment by manniL** [comment]
cc @NicoPennec

File: src/index.js [hunk 1]
```diff
@@ -21,109 +21,121 @@ const defaults = {
...
+  // Merge the options with the default values
+  mergeOrPushSitemaps(this.options.sitemap, options)
+  mergeOrPushSitemaps(moduleOptions, options)
 
   // sitemap-routes.json is written to dist dir on build mode
   const jsonStaticRoutesPath = path.resolve(this.options.buildDir, path.join('dist', 'sitemap-routes.json'))
+  let staticRoutes = fs.readJsonSync(jsonStaticRoutesPath, { throws: false })
 
-  // sitemap.xml is written to static dir on generate mode
-  const xmlGeneratePath = path.resolve(this.options.srcDir, path.join(this.options.dir.static || 'static', options.path))
```
Comment (manniL) [inline src/index.js]: Removing `this.options.dir.static` is not a good idea

File: src/index.js [hunk 2]
```diff
@@ -21,109 +21,121 @@ const defaults = {
...
+  options.sitemaps.map((sitemap) => {
+    // sitemap.xml is written to static dir on generate mode
+    const xmlGeneratePath = path.resolve(this.options.srcDir, path.join('static', sitemap.path))
```
Comment (manniL) [inline src/index.js]:
```suggestion
    const xmlGeneratePath = path.resolve(this.options.srcDir, path.join(this.options.dir.static || 'static', sitemap.path))
```

**Comment by amjadkhan896** [comment]
@manniL @NicoPennec This feature is completed or still pending?. Because I also need this feature.
Thanks

File: src/index.js [hunk 3]
```diff
@@ -21,109 +21,121 @@ const defaults = {
...
+        // Save static routes
+        fs.ensureDirSync(path.resolve(this.options.buildDir, 'dist'))
+        fs.writeJsonSync(jsonStaticRoutesPath, staticRoutes)
+
+        // TODO on generate process only and not on build process
+        if (sitemap.generate) {
+          (async () => {
+            // Generate static sitemap.xml
+            const routes = await sitemap.cache.get('routes')
+            const sm = await createSitemap(sitemap, routes)
```
Comment (manniL) [inline src/index.js]: Could that name be more descriptive? :thinking:

**Comment by manniL** [comment]
cc @NicoPennec ☺️

**Comment by manniL** [comment]
FYI: Had to close and reopen the PR to trigger CircleCI

**Comment by samturrell** [comment]
What's this feature waiting on, out of interest?

**Comment by manniL** [comment]
@WilliamDASILVA could you resolve the merge conflits? ☺️

**Comment by maxkobzev** [comment]
Would be great to add option to create sitemapIndex for multiple sitemaps

**Comment by amjadkhan896** [comment]
Hi all,
I appreciate your hard work for making such a nice plugin for us. I would ask is this feature completed or still  under development. Because as @manniL mentioned up i used my code like below.

```

 sitemap: [
        {
            path: '/sitemap.xml',
            cacheTime: 1000 * 60 * 15,
            hostname: "https://elanat.com",
            routes() {
                //console.log(process.env.apiUrl);
                return axios.get('https://api.elanat.com/api/get-sitemap', {
                    httpsAgent: new https.Agent({
                        rejectUnauthorized: false
                    })
                })
                    .then(res => res.data.map(cat => {
                        const path = 'https://elanat.com/' + cat.url
                        // console.log(cat);
                        return {
                            url: path,
                            changefreq: cat.frequency,
                            priority: cat.priority,
                            lastmodISO: cat.lastModData
                        };

                    }));

            }
        },
        {
            path: '/second-sitemap.xml',
            cacheTime: 1000 * 60 * 15,
            hostname: "https://elanat.com",
            routes() {
                //console.log(process.env.apiUrl);
                return axios.get('https://api.elanat.com/api/get-sitemap', {
                    httpsAgent: new https.Agent({
                        rejectUnauthorized: false
                    })
                })
                    .then(res => res.data.map(cat => {
                        const path = 'https://elanat.com/' + cat.url
                        // console.log(cat);
                        return {
                            url: path,
                            changefreq: cat.frequency,
                            priority: cat.priority,
                            lastmodISO: cat.lastModData
                        };

                    }));

            }
        },

    ],
````
But the first sitemap generated was with empty URLs. and the second responds with not found Error.
We are eagerly waiting for this feature.
Thanks Once again.

**Comment by NicoPennec** [comment]
As explain before the current PR does not match my goal of the multi-sitemap. Thanks again WilliamDASILVA for your proposal.

Then, as explain on another issue, a refactoring with multi-sitemap and sitemap-index is WIP and will be ready in short term.

**Comment by amjadkhan896** [comment]
Thanks, @NicoPennec for the reply. Can you please specify which one is the exact PR. So that we go there and informed by the email once completed.

**Comment by papakay** [comment]
Hi @NicoPennec please any update regarding the multiple sitemaps?

**Pull request closed without merging**